java-checks/src/main/java/org/sonar/java/checks/IfElseIfStatementEndsWithElseCheck.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import java.util.Collections;
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.tree.IfStatementTree;
+import org.sonar.plugins.java.api.tree.StatementTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+@Rule(key = "S126")
+public class IfElseIfStatementEndsWithElseCheck extends IssuableSubscriptionVisitor {
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return Collections.singletonList(Tree.Kind.IF_STATEMENT);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    StatementTree elseStmt = ((IfStatementTree) tree).elseStatement();
+    if (elseStmt != null
+      && elseStmt.is(Tree.Kind.IF_STATEMENT) && ((IfStatementTree) elseStmt).elseStatement() == null) {
+      reportIssue(elseStmt, "\"if ... else if\" constructs should end with \"else\" clauses.");
```

Comment:
You should **not** report on the whole `else` statement. It is going to do way too much noise in the SonarQube UI, and put the issue on the last line, while we want it on the last `else if` line.

Please report only on `else if` (the `else` keyword is available as a token in the parent `IfStatementTree`, and the `if` keyword will be available in the child `ifStatementTree`)
 
Currently, in the following code for instance, it would highlight in red the 8 lines following the `else`!
```java
        if (c == '\'')
          inQuote = true;
        else if ((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z')) {
          int ci = from.indexOf(c);
          if (ci == -1)
            throw new IllegalArgumentException("Illegal pattern " +
              " character '" +
              c + "'");
          c = to.charAt(ci);
        }
      }
```

![image](https://user-images.githubusercontent.com/10588758/37511013-7ab66fcc-28fd-11e8-91d3-94b423a0c6d2.png)